src/map/utils/charutils.cpp

```diff
@@ -4715,6 +4716,29 @@ namespace charutils
         }
     }
 
+    bool IsAidBlocked(CCharEntity* PInitiator, CCharEntity* PTarget) {
+        if (PTarget->getBlockingAid())
+        {
+            bool inAlliance = false;
+            PTarget->ForAlliance([&PInitiator, &PTarget, &inAlliance](CBattleEntity* PEntity) {
+                if (PEntity->id == PInitiator->id)
+                {
+                    inAlliance = true;
+                }
+            });
+
+            if (!inAlliance)
+            {
+                // Target is blocking assistance
+                PInitiator->pushPacket(new CMessageSystemPacket(0, 0, 225));
```

Comment:
There are 2 error messages to different clients, I'm not sure how to send both through IsValidTarget.